### ISSUE TITLE
what about making the resulting type readonly ?

### ISSUE THREAD

**Issue by yavuzmester** [rangle/typed-immutable-record#26]
Hello,
The resulting type after recordify does not have readonly property, thus when we want to write to a field, typescript does not give compile time error but we get runtime error.

`import {TypedRecord, recordify} from 'typed-immutable-record';

interface IAnimal {
    type: string;
    age: number;
}

interface IAnimalRecord extends TypedRecord<IAnimalRecord>, IAnimal {}

const dogRecord = recordify<IAnimal, IAnimalRecord>({
    type: 'Dog',
    age: 5
});

dogRecord.age = 10;  //this line I am talking about`

I can add the readonly flag to the IAnimal interface but I think the resulting type should have it, as records are immutable theirs attributes should be readonly.

**Comment by yavuzmester** [comment]
And the recordify usage is a bit odd, as we have to provide both IAnimalRecord and IAnimal instead of just
`recordify<IAnimalRecord>`

I am new to Typescript, are these limitations caused by Typescript?

**Comment by danielfigueiredo** [comment]
Good idea, sorry about the delay I was out on vacations until yesterday
I will look into this, with the newer version/ features of typescript I have like 3 new awesome enhancements to do
I'm afraid I won't be able to touch this until end of the year since I will be out of the office working on a client
feel free to submit a PR if you need this before the end of this month I can get it merged and publish a new release with it =]

**Comment by danielfigueiredo** [comment]
Just a quick question, could you please tell me an example property that the TypedRecord should have as readonly? The reason I ask is because the properties we dot access are  not declared int he TypedRecord interface
